File: Programmers/소수만들기/solution2.py
def solution(nums):
    def check(num):
        if num > 1:
            for i in range(2, num):
                if num % i == 0:
                    return False
            return True
        else:
            return False
    count, n = 0, len(nums)
    for i in range(n):
        for j in range(i+1, n):
            for k in range(j+1, n):
                if check(nums[i] + nums[j] + nums[k]):
                    count += 1
    return count

a = [1,2,3,4]
print(solution(a))

File: Leet Code/Wiggle Subsequence/solution2.swift
// Test Cases
// [2, 1] -> 2
// [1, 2] -> 2
// [1, 1] -> 1
// [1] -> 1
// missed case
// [3, 3, 3, 2, 5]

class Solution {
    func wiggleMaxLength(_ nums: [Int]) -> Int {
        //edge case
        if nums.count == 1 {
            return nums.count
        }
        
        var currentState: Bool? = nil
        var wiggleCount = 0
        
        for i in 1..<nums.count {
            if nums[i] == nums[i-1] {
                continue
            }
            let isPositive = nums[i] - nums[i-1] > 0
            
            if isPositive == currentState {
                continue
            } else {
                currentState = isPositive
                wiggleCount += 1
            }
        }
        
        return wiggleCount + 1
    }
}
